Failed to run TF mnist example in GCP

Unable to run mnist example in Dataproc.

```
sudo java -cp `hadoop classpath`:/usr/local/src/MyJob/tony-cli-0.1.5-all.jar com.linkedin.tony.cli.ClusterSubmitter --python_venv=/usr/lo
cal/src/MyJob/venv.zip --src_dir=/usr/local/src/TonY/mnist/ --executes=/usr/local/src/TonY/mnist/src/mnist_distributed.py --conf_file=/usr/local/src/tony.xml --python_binary_path
=venv/bin/python3.5
18/11/11 08:23:02 INFO cli.ClusterSubmitter: Starting ClusterSubmitter..
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/usr/lib/hadoop/lib/slf4j-log4j12-1.7.25.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/local/src/MyJob/tony-cli-0.1.5-all.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
18/11/11 08:23:02 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
18/11/11 08:23:02 INFO cli.ClusterSubmitter: Configuration: core-default.xml, core-site.xml, mapred-default.xml, mapred-site.xml, yarn-default.xml, yarn-site.xml, null/core-site.
xml, null/hdfs-site.xml
Nov 11, 2018 8:23:02 AM com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystemBase <clinit>
INFO: GHFS version: hadoop2-1.9.8
18/11/11 08:23:03 WARN shortcircuit.DomainSocketFactory: The short-circuit local reads feature cannot be used because libhadoop cannot be loaded.
18/11/11 08:23:03 INFO cli.ClusterSubmitter: Copying /usr/local/src/MyJob/tony-cli-0.1.5-all.jar to: hdfs://tony-dev-m/user/root/.tony/6665ca2a-fd31-4f61-a947-33f895517302
Exception in thread "main" java.lang.ClassCastException: [Ljava.lang.Object; cannot be cast to [Ljava.lang.String;
        at com.linkedin.tony.cli.ClusterSubmitter.main(ClusterSubmitter.java:60)
```

```
hadoop version
Hadoop 2.9.0
Subversion Unknown -r Unknown
Compiled by bigtop on 2018-08-17T12:00Z
Compiled with protoc 2.5.0
From source with checksum f510b6e8bafb2ddfd660aeb7454e7c30
This command was run using /usr/lib/hadoop/hadoop-common-2.9.0.jar
```
Java version
```
java -version
openjdk version "1.8.0_181"
OpenJDK Runtime Environment (build 1.8.0_181-8u181-b13-1~deb9u1-b13)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)
```
Command run:
```
java -cp `hadoop classpath`:/usr/local/src/MyJob/tony-cli-0.1.5-all.jar com.linkedin.tony.cli.ClusterSubmitter \
--python_venv=/usr/local/src/MyJob/venv.zip \
--src_dir=/usr/local/src/TonY/mnist/ \
--executes=/usr/local/src/TonY/mnist/src/mnist_distributed.py \
--conf_file=/usr/local/src/tony.xml \
--python_binary_path=venv/bin/python3.5
```

Directory structure:

```
.
├── src
│   └── mnist_distributed.py
├── tony-cli-0.1.5-all.jar
├── tony.xml
└── venv.zip
```

tony.xml contents:

```
<configuration>
  <property>
    <name>tony.application.security.enabled</name>
   <value>false</value>
  </property>    
  <property>
    <name>tony.worker.instances</name>
    <value>2</value>
  </property>
  <property>
    <name>tony.worker.memory</name>
    <value>15g</value>
  </property>
  <property>
    <name>tony.worker.gpus</name>
    <value>0</value>
  </property>
  <property>
    <name>tony.ps.memory</name>
    <value>3g</value>
  </property>
</configuration>
```

Yes, other MR job works fine. I tried a sample GCP (wordcount) and worked fine. 

![image](https://user-images.githubusercontent.com/30065079/48331351-b5d9af00-e604-11e8-9334-c8d0eda1dfd2.png)

I was running using the `java -cp hadoop`, but found a GCP command:` gcloud submit hadoop` and now I'm able to submit the job, and get the following:

```
gcloud dataproc jobs submit hadoop --cluster tony-dev \
>   --class com.linkedin.tony.cli.ClusterSubmitter \
>    --jars file:///usr/local/src/jobs/MyJob/tony-cli-0.1.5-all.jar -- \
>   --executes=/usr/local/src/jobs/MyJob/src/mnist_distributed.py \
>   --python_venv=/usr/local/src/deps/venv/venv.zip \
>   --src_dir=/usr/local/src/jobs/MyJob/ \
>   --conf_file=/usr/local/src/jobs/MyJob/tony.xml \
>   --python_binary_path=tony/bin/python3.5
Job [9db641675f914f65baacb2098537d687] submitted.
Waiting for job output...
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/usr/lib/hadoop/lib/slf4j-log4j12-1.7.25.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/usr/local/src/jobs/MyJob/tony-cli-0.1.5-all.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
18/11/12 06:44:04 INFO cli.ClusterSubmitter: Starting ClusterSubmitter..
18/11/12 06:44:05 INFO cli.ClusterSubmitter: Configuration: core-default.xml, core-site.xml, mapred-default.xml, mapred-site.xml, yarn-default.xml, yarn-site.xml, 9db641675f914f65baa
cb2098537d687-conf.xml, /etc/hadoop/conf/core-site.xml, /etc/hadoop/conf/hdfs-site.xml
18/11/12 06:44:05 INFO gcs.GoogleHadoopFileSystemBase: GHFS version: hadoop2-1.9.8
18/11/12 06:44:05 INFO cli.ClusterSubmitter: Copying /usr/local/src/jobs/MyJob/tony-cli-0.1.5-all.jar to: hdfs://tony-dev-m/user/root/.tony/1661f515-4711-4030-9a82-6da5c16fb363
18/11/12 06:44:06 INFO tony.TonyClient: TonY heartbeat interval [1000]
18/11/12 06:44:06 INFO tony.TonyClient: TonY max heartbeat misses allowed [25]
18/11/12 06:44:06 INFO tony.TonyClient: Starting client..
18/11/12 06:44:06 INFO client.RMProxy: Connecting to ResourceManager at tony-dev-m/10.128.0.2:8032
18/11/12 06:44:06 INFO client.AHSProxy: Connecting to Application History server at tony-dev-m/10.128.0.2:10200
18/11/12 06:44:17 INFO tony.TonyClient: Completed setting up Application Master command {{JAVA_HOME}}/bin/java -Xmx1638m -Dyarn.app.container.log.dir=<LOG_DIR> com.linkedin.tony.Tony
ApplicationMaster --python_binary_path tony/bin/python3.5 --python_venv /usr/local/src/deps/venv/venv.zip --executes /usr/local/src/jobs/MyJob/src/mnist_distributed.py --hdfs_classpa
th hdfs://tony-dev-m/user/root/.tony/1661f515-4711-4030-9a82-6da5c16fb363 --container_env TONY_CONF_PATH=hdfs://tony-dev-m/user/root/.tony/application_1541371308113_0012/tony-final.x
ml --container_env TONY_CONF_TIMESTAMP=1542005057940 --container_env TONY_CONF_LENGTH=3523 --container_env TONY_ZIP_PATH=hdfs://tony-dev-m/user/root/.tony/application_1541371308113_0
012/tony.zip --container_env TONY_ZIP_LENGTH=232431960 --container_env TONY_ZIP_TIMESTAMP=1542005057881 --container_env CLASSPATH={{CLASSPATH}}<CPS>./*<CPS>$HADOOP_CONF_DIR<CPS>$HADO
OP_COMMON_HOME/*<CPS>$HADOOP_COMMON_HOME/lib/*<CPS>$HADOOP_HDFS_HOME/*<CPS>$HADOOP_HDFS_HOME/lib/*<CPS>$HADOOP_MAPRED_HOME/*<CPS>$HADOOP_MAPRED_HOME/lib/*<CPS>$HADOOP_YARN_HOME/*<CPS
>$HADOOP_YARN_HOME/lib/*' 1><LOG_DIR>/amstdout.log 2><LOG_DIR>/amstderr.log
18/11/12 06:44:17 INFO tony.TonyClient: Submitting YARN application
18/11/12 06:44:18 INFO impl.YarnClientImpl: Submitted application application_1541371308113_0012
18/11/12 06:44:18 INFO tony.TonyClient: URL to track running application (will proxy to TensorBoard once it has started): http://tony-dev-m:8088/proxy/application_1541371308113_0012/
18/11/12 06:44:18 INFO tony.TonyClient: ResourceManager web address for application: http://tony-dev-m:8088/cluster/app/application_1541371308113_0012
18/11/12 06:44:21 INFO tony.TonyClient: Application 12 finished with YarnState=FAILED, DSFinalStatus=FAILED, breaking monitoring loop.
18/11/12 06:44:21 ERROR tony.TonyClient: Application failed to complete successfully
ERROR: (gcloud.dataproc.jobs.submit.hadoop) Job [9db641675f914f65baacb2098537d687] entered state [ERROR] while waiting for [DONE].

```

--- a/tony-cli/src/main/java/com/linkedin/tony/cli/ClusterSubmitter.java
+++ b/tony-cli/src/main/java/com/linkedin/tony/cli/ClusterSubmitter.java
@@ -57,7 +57,7 @@ public class ClusterSubmitter {
       List<String> updatedArgs = new LinkedList<>(Arrays.asList(args));
       updatedArgs.add(0, cachedLibPath.toString());
       updatedArgs.add(0, "--hdfs_classpath");
-      exitCode = TonyClient.start((String[])updatedArgs.toArray());
+      exitCode = TonyClient.start(updatedArgs.toArray(new String[0]));
     } catch (IOException e) {
       LOG.fatal("Failed to create FileSystem: ", e);
       exitCode = -1;

